test/units/galaxy/test_collection_install.py

```diff
@@ -165,7 +165,7 @@ def test_build_requirement_from_path(collection_artifact):
     assert actual.dependencies == {}
 
 
-@pytest.mark.parametrize('version', ['1.1.1', 1.1, 1])
+@pytest.mark.parametrize('version', ['1.1.1', '1.1.0', '1.0.0'])
```

Comment:
I'm pretty sure this test with floats and integers instead of strings was testing to make sure `MANIFEST.json` with those values instead of a string didn't traceback ansible-galaxy like it did before a commit that fixed it. While `ansible-galaxy collection build` would never create a MANIFEST like that I can't guarantee other build tools would be the same.

Reply:
I'll have to look at the code flow, but when I swapped out `LooseVersion` for `SemanticVersion`, this test began failing.

Maybe I was overzealous in my replacement.

Reply:
This is due to the change at https://github.com/ansible/ansible/pull/68258/files#diff-2de3c8cc3e7fae814d02b3058f0c8f9eR365

That ensures that we can compare the version later with a `SemanticVersion`, so it either needs to stay, and cause us to fall back to `*`, or we would need to do what we do with user requirements:

```
SemanticVersion.from_loose_version(LooseVersion(version))
```

`from_loose_version` takes a simplified version, and attempts to convert it to a `SemanticVersion`.